server/src/immich/app.service.ts

```diff
@@ -22,9 +61,22 @@ export class AppService {
     this.storageService.init();
     await this.searchService.init();
     this.logger.log(`Feature Flags: ${JSON.stringify(await this.serverService.getFeatures(), null, 2)}`);
+
+    const config = await this.configCore.getConfig();
+    if (config.checkAvailableVersion.enabled) {
+      await this.systemConfigService.handleImmichLatestVersionAvailable();
+      const interval = setInterval(
+        () => this.systemConfigService.handleImmichLatestVersionAvailable(),
+        checkIntervalTime,
+      );
+      this.configCore.schedulerRegistry.addInterval('check-available-version', interval);
+    }
   }
 
   async destroy() {
     this.searchService.teardown();
+    if (this.configCore.schedulerRegistry.doesExist('interval', 'check-available-version')) {
+      this.configCore.schedulerRegistry.deleteInterval('check-available-version');
+    }
```

Comment:
Search service uses an interval the same way you should use it. No logic should exist here, just call `service.init()` and `service.destroy()` and move the logic into `domain/`.